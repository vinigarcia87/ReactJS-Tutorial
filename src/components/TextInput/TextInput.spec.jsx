import React from "react";
import {render, screen} from "@testing-library/react";
import {TextInput} from "./index";
import userEvent from "@testing-library/user-event";

describe('<TextInput />', () => {
    it('Should have a value of searchValue', () => {
        const fn = jest.fn();
        render(<TextInput handleChange={fn} searchValue={'testando'}/>);

        const input = screen.getByPlaceholderText(/type your search/i);
        expect(input).toBeInTheDocument();
        expect(input.value).toBe('testando');
    });

    it('Should call handleChange function on each key pressed', async () => {
        const user = userEvent.setup();
        const fn = jest.fn();
        render(<TextInput handleChange={fn}/>);

        const input = screen.getByPlaceholderText(/type your search/i);
        const value = 'o valor';

        await user.type(input, value);

        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length);
    });

    it('Should match snapshot', () => {
        const fn = jest.fn();
        const {container} = render(<TextInput handleChange={fn} searchValue={'testando'}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
})