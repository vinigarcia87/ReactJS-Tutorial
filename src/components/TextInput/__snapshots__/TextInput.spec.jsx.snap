// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should match snapshot 1`] = `
<input
  class="text-input form-control"
  placeholder="Type your search"
  type="search"
  value="testando"
/>
`;
